doc/git.md

```diff
@@ -0,0 +1,167 @@
+*fork* the official (that is, https://www.github.com/IBAMR/IBAMR) IBAMR
+repository by clicking on the Fork button in the top right. This will create a
+separate copy of the *IBAMR/IBAMR* repository in your own GitHub account. After
+doing this, you can *clone* the repository to your local computer in the same
+way as was done for *Spoon-Knife*.
+
+Once you have a local copy of IBAMR you can start making changes. With a
+brand-new repository one usually needs to set their authorship information:
+
+> git config --local user.name "John Doe"
+> git config --local user.email "john.doe@lab.gov"
+
+It is usually helpful to tell the git repository on your computer about a few
+other git repositories: these are called *remotes*. The most important remote is
+your fork on GitHub (which is, itself, another git repository). The convention
+is to call this remote the *origin*. Similarly, the official IBAMR repository on
+GitHub is, by convention, called the *upstream* remote. If you cloned your GitHub
+fork then that fork should be set as your origin:
+
+```
+> git remote show origin
+* remote origin
+  Fetch URL: https://github.com/john.doe/ibamr
+  Push  URL: https://github.com/john.doe/ibamr
+  HEAD branch: master
+  [...]
+```
+if this isn't set up, then try running
+
+> git remote add origin https://github.com/john.doe/ibamr
+
+Use the same command to set up upstream:
+
+> git remote add upstream https://github.com/ibamr/ibamr
+
+That should be all the configuration your local IBAMR repository needs: at this
+point your authorship information is set up and the repository is correctly
+linked with your copy of IBAMR on GitHub and the official IBAMR repository.
+
+Suppose that you wanted to add another function to
+`ibtk/src/lagrangian/LEInteractor.cpp`. You should start by creating a branch
+with a relevant name. If we added a function that calculated the number of
+interaction points, for example, we should name it something like
+
+> git checkout -b add-leinteractor-count-ib-points
+
+Once we are on the branch and have made changes we can stage changes by
+executing
+
+> git add ibtk/src/lagrangian/LEInteractor.cpp
+
+and then commit them with a brief but explanatory message:
+
+> git commit -m "Added a function for counting IB points."
+
+You can inspect the commit by viewing the commit log:
+
+> git log
+
+which will show the commit message, commit hash, time and date, and authorship
+information. Once this is done, one can put the commit on GitHub by running
+
+> git push origin add-leinteractor-count-ib-points
+
+and, navigating to https://github.com/ibamr/ibamr, one can then open a pull
+request by hitting the button `Compare & Pull Request`.
+
+### Cleaning up history on a branch
+Sometimes its necessary to take a sequence of commits on a branch (here
+`feature-branch`) and reorder, reword, or recombine them in some way. The best
+way to do this is to do an *interactive rebase*:
+
+> git checkout feature-branch
+> git rebase -i master
+
```

Comment:
@drwells : These two lines show up on the same line when viewed on Github.

Reply:
Thanks for checking this. I'll put up a fix.